ci/e2e-tests/test-run.sh

```diff
@@ -103,25 +118,48 @@ get_package() {
     artifact_name="packages_${artifact_name}_amd64"
 
     echo 'Getting artifact download URL...' >&2
-    artifact_download_url="$(
-        github_curl -L \
-            -H 'accept: application/vnd.github.v3+json' \
-            "$artifacts_url" |
-            jq \
-                --arg artifact_name "$artifact_name" \
-                -r \
-                '.artifacts[] | select(.name == $artifact_name) | .archive_download_url'
-    )"
+    artifact_download_url=""
+
+    for retry in {0..3}; do
+        if [ "$retry" != '0' ]; then
+            sleep 10
+        fi
+
+        artifact_download_url="$(
+            github_curl -L \
+                -H 'accept: application/vnd.github.v3+json' \
+                "$artifacts_url" |
+                jq \
+                    --arg artifact_name "$artifact_name" \
+                    -r \
+                    '.artifacts[] | select(.name == $artifact_name) | .archive_download_url'
+        )"
+
+        if [ ! -z "$artifact_download_url" ]; then
+            break
+        fi
+    done
+
     if [ -z "$artifact_download_url" ]; then
         echo "Could not find artifact for OS $OS" >&2
         exit 1
     fi
     echo "Artifact download URL: $artifact_download_url" >&2
 
     echo 'Downloading artifact...' >&2
-    github_curl -L \
-        -o package.zip \
-        "$artifact_download_url"
+    for retry in {0..3}; do
+        if [ "$retry" != '0' ]; then
+            sleep 10
+        fi
+
+        github_curl -L \
+            -o package.zip \
+            "$artifact_download_url"
+
+        if [ -f package.zip ]; then
```

Comment:
This will pass if the download aborts in between and you're left with an incomplete file. You should check the exit code of `github_curl` instead.

Reply:
But it will fail to unzip an incomplete file immediately after this. And with set e, the script will abort if curl fails, so we can assume the exit code was 0 by this line, right?

Reply:
If curl succeeds and unzip fails, that just means even the unzip needs to be inside the retry.

Reply:
Fixed